packages/flutter/lib/src/cupertino/switch.dart

```diff
@@ -501,18 +501,16 @@ class _RenderCupertinoSwitch extends RenderConstrainedBox {
       visualPosition,
     );
     final double thumbCenterY = offset.dy + size.height / 2.0;
-
-    canvas.save();
-    canvas.clipRRect(trackRRect);
-    canvas.saveLayer(trackRect, Paint());
-    _thumbPainter.paint(canvas, Rect.fromLTRB(
+    final thumbBounds = Rect.fromLTRB(
       thumbLeft,
       thumbCenterY - CupertinoThumbPainter.radius,
       thumbRight,
       thumbCenterY + CupertinoThumbPainter.radius,
-    ));
-    canvas.restore();
-    canvas.restore();
+    );
+
+    context.pushClipRRect(needsCompositing, Offset.zero, thumbBounds, trackRRect, (innerContext, _) {
```

Comment:
I wasn't sure whether to shorten this line and I'm hesitant to run dartfmt on library source (though if that's standard practice, I'll do that and re-push)

Reply:
The analyzer isn't happy about this line. See https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#naming-rules-for-typedefs-and-function-variables
> If you have a callback with arguments but you want to ignore the arguments, give the type and names of the arguments anyway. That way, if someone copies and pastes your code, they will not have to look up what the arguments are.


```suggestion
    context.pushClipRRect(needsCompositing, Offset.zero, thumbBounds, trackRRect, (PaintContext innerContext, Offset offset) {
```

Reply:
Regarding dartfmt: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#formatting

> These guidelines have no technical effect, but they are still important purely for consistency and readability reasons.
> We do not yet use dartfmt. Flutter code tends to use patterns that the standard Dart formatter does not handle well. We are working with Dart team to make dartfmt aware of these patterns.